core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -443,13 +476,10 @@ class ReplicaManager(val config: KafkaConfig,
         } catch {
           // NOTE: Failed produce requests metric is not incremented for known exceptions
           // it is supposed to indicate un-expected failures of a broker in handling a produce request
-          case e: KafkaStorageException =>
-            fatal("Halting due to unrecoverable I/O error while handling DeleteRecordsRequest: ", e)
-            Runtime.getRuntime.halt(1)
-            (topicPartition, null)
           case e@ (_: UnknownTopicOrPartitionException |
                    _: NotLeaderForPartitionException |
                    _: OffsetOutOfRangeException |
+                   _: KafkaStorageException |
```

Comment:
I still keep the code to catch `KafkaStorageException` since it may be useful if the underlying code re-throws KafkaStorageException in the future. It is a safe choice. But we no longer needs to call `maybeAddLogFailureEvent` when KafkaStorageException is caught for the reasons explained in the other comment.